### ISSUE TITLE
What is the Expected Inference Performance

### ISSUE THREAD

**Issue by tcapelle** [ml-explore/mlx#49]
I am running Llama/Mistral inference examples on my M1Pro with 16GB of memory and getting around 80sec/token. 
- Does the framework support FP16?
- GPU usage seems low, do I need to do something to use the Metal GPU?
- `mx.default_device` reports `Device(gpu, 0)`
<img width="259" alt="image" src="https://github.com/ml-explore/mlx/assets/18441985/014fa431-14ba-41a2-93c2-572ebb1b0bee">

**Comment by tcapelle** [comment]
I am curious if someone managed to run this on a laptop outside of the Ultras.

**Comment by lin72h** [comment]
> I am running Llama/Mistral inference examples on my M1Pro with 16GB of memory and getting around 80sec/token.
Are you using the 7b llama and 7b mistral model? Is it a typo? Do you mean(80ms/token or 80sec/token)?

**Comment by arpan-dhatt** [comment]
@tcapelle Can you please check your memory pressure when running the model? At 16GB of memory, you may be running out of wired memory since the example uses FP16 (weights total nearly 14.6GB) and inference takes a bit more than that.

**Comment by briancpark** [comment]
@tcapelle I tried the LLaMA example on my M1 Pro 32GB. It's indeed slow, and I think that's mostly due to the weights being FP32. I haven't checked Mistral example yet, but this performance is expected if that is also FP32. Transformer inference is typically memory-bound and using FP32 is a bottleneck. 

Did you do additional modifications to run the example in FP16 or did I miss something?

**Comment by rovo79** [comment]
Running Inference with MLX won't touch ANE in anyway, right?

**Comment by briancpark** [comment]
@rovo79 Correct. As per #18, ANE API is closed source and not publicly accessible. I believe the only way to touch ANE today is via CoreML.

**Comment by tcapelle** [comment]
So, has someone managed to run a 7B inference using MLX on 16GB of Ram?

**Comment by zcbenz** [comment]
This issue was about inference performance about particular models, which I don't think is relevant today and I'm closing it. If you observed performance issues with new models, please open new issues.
